src/numeric_index.c

```diff
@@ -810,3 +811,46 @@ void NumericRangeTreeIterator_Free(NumericRangeTreeIterator *iter) {
   array_free(iter->nodesStack);
   rm_free(iter);
 }
+
+/* A callback called after a concurrent context regains execution context. When this happen we need
+ * to make sure the key hasn't been deleted or its structure changed, which will render the
+ * underlying iterators invalid */
+void NumericRangeIterator_OnReopen(void *privdata) {
+  NumericUnionCtx *nu = privdata;
+  IndexSpec *sp = nu->sp;
```

Comment:
I believe the right approach would be to reopen the spec here then check the revisionId and:
1. Short solution - abort
2. Long term solution - Been able to continue the iteration somehow ...

Reply:
About long term, if we allow updates, new entries could be inserted before or after the current cursor, so either some preceding values might not be returned by the cursor, or if rewinded, some values might be returned more than once.

Reply:
We are not doing it for either `IndexReader_OnReopen` and `TagReader_OnReopen`. Why is it necessary now?

Reply:
We do continue from the same docid that we stopped: https://github.com/RediSearch/RediSearch/blob/ba041f443ae4bbb02696a4099036e3bbd049778b/src/inverted_index.c#L135

Reply:
Here we have two nodes instead of one.